lib/src/default_index_store.rs

```diff
@@ -559,8 +559,7 @@ impl MutableIndexImpl {
             buf.write_u32::<LittleEndian>(pos.0).unwrap();
         }
 
-        buf[parent_overflow_offset..parent_overflow_offset + 4]
-            .as_mut()
+        (&mut buf[parent_overflow_offset..parent_overflow_offset + 4])
```

Comment:
Not related to this change:

Might be just me, but I personally prefer `buf[parent_overflow_offset..][..4]` to avoid having to use `parent_overflow_offset` twice

Reply:
Nice, I like that trick. I'll try to remember to send a PR for that after this one's in.

Reply:
I guess this doesn't block the current PR, resolving.